scipy/signal/tests/test_filter_design.py

```diff
@@ -826,34 +832,34 @@ def test_fs_param(self):
         # N = None, whole=False
         w1, h1 = freqz(b, a, fs=fs)
         w2, h2 = freqz(b, a)
-        assert_allclose(h1, h2)
-        assert_allclose(w1, np.linspace(0, fs/2, 512, endpoint=False))
+        xp_assert_close(h1, h2)
+        xp_assert_close(w1, np.linspace(0, fs/2, 512, endpoint=False))
 
         # N = None, whole=True
         w1, h1 = freqz(b, a, whole=True, fs=fs)
         w2, h2 = freqz(b, a, whole=True)
-        assert_allclose(h1, h2)
-        assert_allclose(w1, np.linspace(0, fs, 512, endpoint=False))
+        xp_assert_close(h1, h2)
+        xp_assert_close(w1, np.linspace(0, fs, 512, endpoint=False))
 
         # N = 5, whole=False
         w1, h1 = freqz(b, a, 5, fs=fs)
         w2, h2 = freqz(b, a, 5)
-        assert_allclose(h1, h2)
-        assert_allclose(w1, np.linspace(0, fs/2, 5, endpoint=False))
+        xp_assert_close(h1, h2)
+        xp_assert_close(w1, np.linspace(0, fs/2, 5, endpoint=False))
 
         # N = 5, whole=True
         w1, h1 = freqz(b, a, 5, whole=True, fs=fs)
         w2, h2 = freqz(b, a, 5, whole=True)
-        assert_allclose(h1, h2)
-        assert_allclose(w1, np.linspace(0, fs, 5, endpoint=False))
+        xp_assert_close(h1, h2)
+        xp_assert_close(w1, np.linspace(0, fs, 5, endpoint=False))
 
         # w is an array_like
         for w in ([123], (123,), np.array([123]), (50, 123, 230),
                   np.array([50, 123, 230])):
             w1, h1 = freqz(b, a, w, fs=fs)
             w2, h2 = freqz(b, a, 2*pi*np.array(w)/fs)
-            assert_allclose(h1, h2)
-            assert_allclose(w, w1)
+            xp_assert_close(h1, h2)
+            xp_assert_close(np.asarray(w), w1, check_dtype=False)
```

Comment:
I'm scared of being too nitpicky, but again the incorrect ordering of `actual` and `desired` causes confusion here - we shouldn't need to coerce the output of the function before passing as `actual`, it should be the immediate result.

If you wouldn't like to make the change here, I could do it in a follow-up.